app/views/assignments/index.html.erb

```diff
@@ -16,29 +27,22 @@
       <%= print_date(week) %>
     </td>
     
-    <% week_assignments = week.employees.order(:first_name) %>
-    <% week_assignments.each do |employee| %>
+    <% week_assignments = get_weekly_office_assignments(week, @office) %>
+    <% week_assignments.each do |assignment| %>
```

Comment:
Line 30 will kick off a different query for every week creating N+1 queries. Instead, assignments should be preloaded. A week should probably belong to an office (each office might have a different schedule, who knows). Then, this can just be `week.assignments.each` and in the controller, you'd want to load the weeks and include assignments (something like this: `@week = Week.includes(:assignments).where(office: current_office)`).

http://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-includes